unity-renderer/Assets/Features/ScreenshotCamera/Scripts/ScreenshotCamera.cs

```diff
@@ -0,0 +1,173 @@
+    {
+        [Header("EXTERNAL DEPENDENCIES")]
+        [SerializeField] private DCLCharacterController characterController;
+        [SerializeField] private CameraController cameraController;
+
+        [Header("MAIN COMPONENTS")]
+        [SerializeField] private Camera cameraPrefab;
+        [SerializeField] private ScreenshotHUDView screenshotHUDViewPrefab;
+
+        [Header("INPUT ACTIONS")]
+        [SerializeField] private InputAction_Trigger cameraInputAction;
+        [SerializeField] private InputAction_Trigger takeScreenshotAction;
+
+        private bool isInstantiated;
+        private bool isInScreenshotMode;
+
+        private Camera screenshotCamera;
+        private ScreenshotHUDView screenshotHUDView;
+
+        private Transform characterCameraTransform;
+        private IAvatarsLODController avatarsLODControllerLazyValue;
+
+        private ScreenshotCapture screenshotCaptureLazyValue;
+        private ICameraReelNetworkService cameraReelNetworkServiceLazyValue;
+
+        private bool prevUiHiddenState;
+        private bool prevMouseLockState;
+        private bool prevMouseButtonCursorLockMode;
+        private bool? isGuestLazyValue;
+
+        private IAvatarsLODController avatarsLODController => avatarsLODControllerLazyValue ??= Environment.i.serviceLocator.Get<IAvatarsLODController>();
+        private ICameraReelNetworkService cameraReelNetworkService => cameraReelNetworkServiceLazyValue ??= Environment.i.serviceLocator.Get<ICameraReelNetworkService>();
+        private bool isGuest => isGuestLazyValue ??= UserProfileController.userProfilesCatalog.Get(player.ownPlayer.Get().id).isGuest;
+
+        private DataStore_Player player => DataStore.i.player;
+
+        private FeatureFlag featureFlags => DataStore.i.featureFlags.flags.Get();
+
+        private ScreenshotCapture screenshotCapture
+        {
+            get
+            {
+                if (isInstantiated)
+                    return screenshotCaptureLazyValue;
+
+                InstantiateCameraObjects();
+
+                return screenshotCaptureLazyValue;
+            }
+        }
+
+        // TODO(Vitaly): Remove this logic when feature flag will be enalbed
+        private IEnumerator Start()
+        {
+            enabled = false;
+            yield return new WaitUntil(() => featureFlags.IsInitialized);
+
+            if (!featureFlags.IsFeatureEnabled("camera_reel"))
+                Destroy(gameObject);
+            else
+                enabled = true;
+        }
+
+        private void OnEnable()
+        {
+            cameraInputAction.OnTriggered += ToggleScreenshotCamera;
+            takeScreenshotAction.OnTriggered += CaptureScreenshot;
+        }
+
+        private void OnDisable()
+        {
+            cameraInputAction.OnTriggered -= ToggleScreenshotCamera;
+            takeScreenshotAction.OnTriggered -= CaptureScreenshot;
+        }
+
+        private void ToggleScreenshotCamera(DCLAction_Trigger _)
```

Comment:
This method does quite a lot of work which I would love to see it being covered by unit tests 🤔

Reply:
A lot of testing is missing. I was thinking to cover with tests on the polishing phase... But do you think it is better to do it now?

Reply:
Polishing phase sounds nice :D